qdrant-landing/content/documentation/fastembed/fastembed-custom-models.md

```diff
@@ -0,0 +1,525 @@
+            "center_crop": False,
+            "mean": [0.485, 0.456, 0.406],
+            "std": [0.229, 0.224, 0.225]
+        }
+        
+        with open(os.path.join(cache_dir, 'config.json'), 'w') as f:
+            json.dump(config, f, indent=2)
+        
+        with open(os.path.join(cache_dir, 'preprocessor_config.json'), 'w') as f:
+            json.dump(preprocessor_config, f, indent=2)
+
+    def embed(
+        self, 
+        images: List[str], 
+        batch_size: int = 16, 
+        parallel: int = None,
+        **kwargs
+    ) -> List[np.ndarray]:
+        embeddings = []
+        
+        for i in range(0, len(images), batch_size):
+            batch = images[i:i+batch_size]
+            processed_images = [self._preprocess_image(img) for img in batch]
+            input_tensor = np.concatenate(processed_images, axis=0)
+            ort_inputs = {self.session.get_inputs()[0].name: input_tensor}
+            batch_embeddings = self.session.run(None, ort_inputs)[0]
+            embeddings.extend(batch_embeddings)
+        
+        return embeddings
+
+    def _preprocess_image(self, image_path: str) -> np.ndarray:
+        image = Image.open(image_path).convert('RGB')
+        tensor = self.transform(image)
+        return tensor.unsqueeze(0).numpy()
+
+    @classmethod
+    def list_supported_models(cls) -> List[Dict[str, Any]]:
+        return [
+            {
+                "model": "blip-image-encoder",
+                "dim": 768,
+                "description": "BLIP Image Embedding Encoder",
+                "license": "apache-2.0",
+                "size_in_GB": 0.5,
+                "sources": {"hf": "rjeeva/blip-image-embedding"},
+                "model_file": "model.onnx"
+            }
+        ]
+```
+## Part 3: Registering and Using Custom Models
+
+### Step 1: Modify ImageEmbedding Initialization
+
+To ensure our custom models work smoothly with FastEmbed, we need to modify the initialization method:
+
+```python
+def modify_image_embedding_init():
+    def custom_init(self, model_name, cache_dir=None, **kwargs):
+        # Remove unsupported kwargs
+        kwargs.pop('threads', None)
+        kwargs.pop('providers', None)
+        kwargs.pop('cuda', None)
+        kwargs.pop('device_ids', None)
+        kwargs.pop('lazy_load', None)
+        
+        # Find the correct embedding class
+        for EMBEDDING_MODEL_TYPE in self.EMBEDDINGS_REGISTRY:
+            supported_models = EMBEDDING_MODEL_TYPE.list_supported_models()
+            if any(model_name.lower() == model['model'].lower() for model in supported_models):
+                self.model = EMBEDDING_MODEL_TYPE(
+                    model_name,
+                    cache_dir,
+                    **kwargs
+                )
+                return
+        
+        raise ValueError(f"Model {model_name} is not supported in ImageEmbedding.")
+    
+    # Monkey patch the __init__ method
+    ImageEmbedding.__init__ = custom_init

```

Comment:
Monkey-patching is an antipattern, and if that's the only way to do it, we need to make FastEmbed easier to extend. @joein What do you think?